index.rst

```diff
@@ -68,9 +68,11 @@ Visit processing can fail at any point, including within and between tasks, and
 Even within a single worker, the order of events is not fully deterministic (e.g., two snaps could be ingested and processed up to snap combination in parallel, though we have not yet tried to do so).
 However, the possible failures can be divided into a handful of cases:
 
-- Any failure before the end of APDB writing (which is atomic as of October 2023) has no consequences beyond that worker's local repository.
+- Any failure before the start of APDB writing (which is not necessarily atomic) has no consequences beyond that worker's local repository.
   The repository may have ingested raws, or partial pipeline outputs.
-- A failure after APDB writing but before the end of ``DiaPipelineTask``, which contains it, is recoverable only if the task can account for the fact that a previous session already wrote to the APDB.
+- A failure before or during DiaObject writing, but before the start of DiaSource writing, is recoverable if DiaObject entries can be updated without conflict.
+  All of a DiaObject's properties, except for its presence, are derived from its associated DiaSources, so no change is permanent until those are written.
```

Comment:
The referent for "those" is a little unclear.  Could be DiaObject properties or associated DiaSources.  I think you mean "those DiaSources", right?